src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

```diff
@@ -45,32 +43,26 @@ function BaseOnboardingAccounting({shouldUseNativeStyles, route}: BaseOnboarding
     // We need to use isSmallScreenWidth, see navigateAfterOnboarding function comment
     // eslint-disable-next-line rulesdir/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth
     const {onboardingIsMediumOrLargerScreenWidth, isSmallScreenWidth, shouldUseNarrowLayout} = useResponsiveLayout();
-    const [onboardingValues] = useOnyx(ONYXKEYS.NVP_ONBOARDING);
     const [onboardingPurposeSelected] = useOnyx(ONYXKEYS.ONBOARDING_PURPOSE_SELECTED);
     const [onboardingPolicyID] = useOnyx(ONYXKEYS.ONBOARDING_POLICY_ID);
-    const [allPolicies, allPoliciesResult] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
     const [onboardingAdminsChatReportID] = useOnyx(ONYXKEYS.ONBOARDING_ADMINS_CHAT_REPORT_ID);
     const [onboardingCompanySize] = useOnyx(ONYXKEYS.ONBOARDING_COMPANY_SIZE);
+    const [allPolicies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const onboardingPolicyForOD = Object.values(allPolicies ?? {}).find(PolicyUtils.isPaidGroupPolicy);
     const {canUseDefaultRooms} = usePermissions();
     const {activeWorkspaceID} = useActiveWorkspace();
 
     const [userReportedIntegration, setUserReportedIntegration] = useState<OnboardingAccounting | undefined>(undefined);
     const [error, setError] = useState('');
-    const isVsb = onboardingValues && 'signupQualifier' in onboardingValues && onboardingValues.signupQualifier === CONST.ONBOARDING_SIGNUP_QUALIFIERS.VSB;
 
-    // If the signupQualifier is VSB, the company size step is skip.
-    // So we need to create the new workspace in the accounting step
+    // Set onboardingPolicyID and onboardingAdminsChatReportID if a workspace is created by the backend for OD signups
     useEffect(() => {
-        const filteredPolicies = Object.values(allPolicies ?? {}).filter(PolicyUtils.isPaidGroupPolicy);
-        if (!isVsb || filteredPolicies.length > 0 || isLoadingOnyxValue(allPoliciesResult)) {
+        if (!onboardingPolicyForOD || onboardingPolicyID) {
             return;
         }
-
-        const {adminsChatReportID, policyID} = Policy.createWorkspace(undefined, true, '', Policy.generatePolicyID(), CONST.ONBOARDING_CHOICES.MANAGE_TEAM);
```

Comment:
If we _were_ creating a policy here I'd assume that maybe we still need to create one until the BE changes are merged.

Reply:
I've added the changes assuming the FE and BE will be merged at the same time. However, if we plan to merge the FE first, I will update the code accordingly.